File: app/src/main/java/at/shockbytes/coins/currency/CurrencyManager.java
package at.shockbytes.coins.currency;

import java.util.List;

import rx.Observable;

/**
 * @author Martin Macheiner
 *         Date: 15.06.2017.
 */

public interface CurrencyManager {

    void setLocalCurrency(Currency currency);

    Currency getLocalCurrency();

    CurrencyConversionRates getCurrencyConversionRates();

    Observable<List<OwnedCurrency>> getOwnedCurrencies();

    OwnedCurrency getOwnedCurrencyById(long id);

    Observable<List<OwnedCurrency>> getCashedoutCurrencies();

    void addOwnedCurrency(OwnedCurrency ownedCurrency);

    Balance getBalance();

    void storeLatestBalance();

    double getLatestBalance();

    void removeCurrency(OwnedCurrency ownedCurrency);

    void cashoutCurrency(OwnedCurrency ownedCurrency);

    void splitCashout(OwnedCurrency currency, double amountToPayout);
}

File: app/src/main/java/at/shockbytes/coins/currency/CurrencyManager.kt
package at.shockbytes.coins.currency

import at.shockbytes.coins.currency.conversion.CurrencyConversionProvider
import io.reactivex.Observable

/**
 * @author Martin Macheiner
 * Date: 15.06.2017.
 */

interface CurrencyManager {

    var localCurrency: RealCurrency

    val ownedCurrencies: Observable<List<Currency>>

    val cashedOutCurrencies: Observable<List<Currency>>

    var balance: Balance

    val latestBalance: Double

    val currencyConversionProvider: CurrencyConversionProvider

    fun getCurrencyById(id: Long): Currency

    fun addCurrency(currency: Currency)

    fun updateBoughtDateCurrency(currency: Currency, boughtDate: Long)

    fun storeLatestBalance()

    fun removeCurrency(ownedCurrency: Currency)

    fun cashoutCurrency(ownedCurrency: Currency)

    fun partialCashout(currency: Currency, amountToPayout: Double)
}
